src/Element/WebformPanoramaTour.php

```diff
@@ -0,0 +1,617 @@
+          ],
+        ];
+        $element['hotspots_temp']['pitch'] = [
+          '#title' => t('Hotspot Pitch'),
+          '#type' => 'textfield',
+          '#size' => '6',
+          '#attributes' => [
+            'data-drupal-loaded-node' => $nodeid,
+            'data-drupal-hotspot-property' => 'pitch',
+          ],
+        ];
+
+        $element['hotspots_temp']['type'] = [
+          '#title' => t('Hotspot Type'),
+          '#type' => 'select',
+          '#options' => [
+            'text' => 'Text',
+            'url' => 'An External URL',
+            'ado' => 'Another Digital Object',
+            //'scene' => 'Another Panorama Scene', LOGIC STILL MISSING!
+          ],
+          '#attributes' => [
+            'data-drupal-loaded-node' => $nodeid,
+            'data-drupal-hotspot-property' => 'type',
+          ],
+        ];
+        $element['hotspots_temp']['label'] = [
+          '#title' => t('Label'),
+          '#type' => 'textfield',
+          '#size' => '12',
+          '#attributes' => [
+            'data-drupal-loaded-node' => $nodeid,
+            'data-drupal-hotspot-property' => 'label',
+          ],
+        ];
+        $element['hotspots_temp']['url'] = [
+          '#title' => t('url'),
+          '#description' => t('Only applies to Hotspots of type "An External URL"'),
+          '#type' => 'url',
+          '#size' => '12',
+          '#attributes' => [
+            'data-drupal-loaded-node' => $nodeid,
+            'data-drupal-hotspot-property' => 'url',
+          ],
+        ];
+
+        $element['hotspots_temp']['ado'] = [
+          '#type' => 'entity_autocomplete',
+          '#title' => t('Select a Digital Object'),
+          '#description' => t('Only applies to Hotspots of type "Another Digital Object"'),
+          '#target_type' => 'node',
+          '#selection_handler' => 'solr_views',
+          '#selection_settings' => [
+            'view' => [
+              'view_name' => 'ado_selection_by_type',
+              'display_name' => 'entity_reference_solr_2',
+              'arguments' => ['Image'],
+            ],
+          ],
+          '#attributes' => [
+            'data-drupal-loaded-node' => $nodeid,
+            'data-drupal-hotspot-property' => 'ado',
+          ],
+        ];
+
+
+        // To make sure menu variables are passed
+        // we limit validation errors to those elements
+        $limit = array_merge($element['#parents'], ['hotspots_temp']);
+
+        $element['hotspots_temp']['add_hotspot'] = [
+          '#type' => 'submit',
+          '#value' => t('Add Hotspot'),
+          '#name' => $element['#name'] . '_addhotspot_button',
+          '#submit' => [[static::class, 'addHotspotSubmit']],
+          '#ajax' => [
+            'callback' => [static::class, 'addHotSpotCallBack'],
+          ],
+          '#button_type' => 'default',
+          '#visible' => 'true',
+          '#limit_validation_errors' => FALSE
```

Comment:
Note to myself. This is affecting the Webform interaction when using them as ingest method. If this element is in any other step that the first, when pressed it resets the step counter (which Webform provides) and we go to step 1.